main.py

```diff
@@ -1,16 +1,153 @@
+import asyncio
+import base64
+import datetime
+import glob
+import json
 import os
+import ssl
+from asyncio import Lock
+
+import certifi
 import decky_plugin
+from aiohttp import ClientSession
 from settings import SettingsManager
 
+
 class Plugin:
+    yt_process: asyncio.subprocess.Process | None = None
+    # We need this lock to make sure the process output isn't read by two concurrent readers at once.
+    yt_process_lock = Lock()
+    music_path = f"{decky_plugin.DECKY_PLUGIN_RUNTIME_DIR}/music"
+    cache_path = f"{decky_plugin.DECKY_PLUGIN_RUNTIME_DIR}/cache"
+    ssl_context = ssl.create_default_context(cafile=certifi.where())
+
     async def _main(self):
-         self.settings = SettingsManager(name="config", settings_directory=decky_plugin.DECKY_PLUGIN_SETTINGS_DIR)
+        self.settings = SettingsManager(
+            name="config", settings_directory=decky_plugin.DECKY_PLUGIN_SETTINGS_DIR
+        )
 
     async def _unload(self):
-        pass
+        if self.yt_process is not None:
+            self.yt_process.terminate()
 
     async def set_setting(self, key, value):
         self.settings.setSetting(key, value)
 
     async def get_setting(self, key, default):
-        return self.settings.getSetting(key, default)
\ No newline at end of file
+        return self.settings.getSetting(key, default)
+
+    async def search_yt(self, term: str):
+        if self.yt_process is not None:
+            self.yt_process.terminate()
```

Comment:
Add `await self.yt_process.wait()` to wait for process to terminate.

Reply:
I'm currently using `await self.yt_process.communicate()` instead of `wait()` per [the recommendation here](https://stackoverflow.com/a/78478115). Since a pipe is attached to capture stdout, if yt-dlp tries to output anything without us reading it, this could get us stuck in a deadlock. Or is there another reason to prefer `wait()` here? I don't have a lot of experience with `asyncio` code yet.